src/app/frontend/common/scaling/scaleresource_controller.js

```diff
@@ -66,47 +60,73 @@ export default class UpdateReplicasDialogController {
     /** @export {!angular.FormController} Initialized from the template */
     this.updateReplicasForm;
 
-    /** @private {!angular.$q.Promise} */
-    this.tokenPromise = kdCsrfTokenService.getTokenForAction('replicationcontroller');
+    /** @export {string} */
+    this.resourceKindName = resourceKindName;
+
+    /** @export {string} */
+    this.resourceName = resourceName;
   }
 
   /**
-   * Updates number of replicas in replication controller.
+   * Updates number of replicas for a resource that is scalable.
    *
    * @export
    */
-  updateReplicas() {
-    if (this.updateReplicasForm.$valid) {
-      this.tokenPromise.then((token) => {
-        let resource = getReplicationControllerSpecPodsResourceWithActions(
-            new StateParams(this.namespace_, this.replicationController), this.resource_,
-            {save: {method: 'POST', headers: {'X-CSRF-TOKEN': token}}});
-
-        /** @type {!backendApi.ReplicationControllerSpec} */
-        let replicationControllerSpec = {
-          replicas: this.replicas,
-        };
-        resource.save(
-            replicationControllerSpec, this.onUpdateReplicasSuccess_.bind(this),
-            this.onUpdateReplicasError_.bind(this));
-      }, this.onUpdateReplicasError_.bind(this));
+  scaleResource() {
+    return this.getResource()
+        .update(this.onUpdateReplicasSuccess_.bind(this), this.onUpdateReplicasError_.bind(this))
+        .$promise;
+  }
+
+  getResource() {
+    return this.resource_(
+        `api/v1/scale/${this.resourceKindName}/${this.namespace_}/${this.resourceName}/`,
+        {'scaleBy': this.desiredPods}, {
+          update: {
+            // redefine update action defaults
+            method: 'PUT',
+            transformRequest: function(headers) {
+              headers = angular.extend({}, headers, {'Content-Type': 'application/json'});
+              return angular.toJson(headers);
+            },
+          },
+        });
+  }
+
+  isJob() {
```

Comment:
Also try to shorten file names, ie. if there is only one module in directory, then you can name it `module.js`. Same with services or controllers.